### PR TITLE
When a user is banned or timedout, all of their messages are removed 

### LINKED ISSUE #48: Deletion of messages after banning the author is not being reflected
When a user first gets banned and then their message gets deleted, the chat view does update accordingly: the deleted message persists.

### Expected behavior
A deleted message should be removed from the chat view, even if its author has been banned before.

### Additional info
According to @BenDMyers, first deleting the message (and then banning the user) works fine.
Observed during maxcellw's twitch stream.

### PR DESCRIPTION
**Pull request by BenDMyers**
Resolves #48 - thanks for the upstream fix in @KenAKAFrosty!

This PR is adapted from code contributed by @KenAKAFrosty in #67, adapted to leverage ESM and to have a cleaner commit history.

## Validation

Go to the [deploy preview](https://deploy-preview-76--showmychat.netlify.app/), and create your own overlay. From an alt account, send messages in a channel. Then, as a mod in that channel, ban the alt account. Verify that the message is removed from the overlay.

### CODE REVIEW

**Comment by NickyMeuleman** [comment]
This is awesome!
Very clean too, thanks to the upstream work

**Comment by KenAKAFrosty** [comment]
Looks amazing! Let's get that bad boy merged 😎

File: src/scripts/utilities.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Removes all messages sent by a given user from the overlay.
+ * Called when a user is banned.
+ *
+ * @param {string} username - account whose messages to remove from the overlay
+ */
+export function removeAllMessagesFromUser(username) {
+	const messagesFromUser = Array.from(
```
[the remark this answers is not in the document]
Comment (BenDMyers) [reply src/scripts/utilities.js:8]: Ooooh, didn't know that! I'll give that a shot and clean up this code for a bit.

**Comment by BenDMyers** [comment]
Merging it in! Thanks for the upstream work and the initial pull request, @KenAKAFrosty!

**Pull request merged**